tests/integration/runtimes/test_network_failures.py

```diff
@@ -62,12 +63,8 @@ def _create_head(port, connection_list_dict, polling, retries=-1):
 def _send_request(gateway_port, protocol):
```

Comment:
I think is better to return and leave the `asserts` to the actual `test` functions so is easier to spot

Reply:
I am not sure how this can be done, since this is often run in its own process (due to reasons of gRPC and pytest not liking each other)